sdk/identity/Azure.Identity/src/Credentials/CredentialOptionsMapper.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Azure.Identity.Credentials;
+
+internal class CredentialOptionsMapper
+{
+    /// <summary>
+    /// Attempts to get broker options and returns them if available, along with a flag indicating success.
+    /// </summary>
+    /// <param name="fileSystem">The file system service to use for reading authentication records.</param>
+    /// <param name="isBrokerEnabled">Output parameter indicating whether broker options are available.</param>
+    /// <returns>The broker options if available, null otherwise.</returns>
+    internal static InteractiveBrowserCredentialOptions TryGetBrokerOptions(out bool isBrokerEnabled, IFileSystemService fileSystem = null)
```

Comment:
do we need this implementation? The one below seems like it would do the same thing when credentialOptions is null.